.github/workflows/build-ubuntu.yaml

```diff
@@ -0,0 +1,59 @@
+name: build-ubuntu
+
+
+# Controls when the action will run
+on:
+  push:
+  pull_request:
+  workflow_dispatch: # trigger the workflow manually
+  # # trigger the workflow on schedule
+  # # https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#scheduled-events
+  # schedule:
+  #   - cron:  '* * * * *'
+
+
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    runs-on: ${{ matrix.runner }}
+    continue-on-error: false
+    strategy:
+      matrix:
+        runner: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        compiler: [clang++, g++]
+        python: [python2, python3]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install requirements
+        run: >
+          sudo apt install --no-install-recommends -y
+          build-essential
+          clang
+          python
+          python-numpy
+          python3
+          python3-numpy
+          cmake
+          gfortran
+          git
+          libopenmpi-dev
```

Comment:
We don't need MPI either, we don't test it below and it can be skipped for this CI. The `openmpi-bin` and `openssh-client` are also part of MPI.